eqs/src/configuration.rs

```diff
@@ -7,9 +7,44 @@
 
 use dirs::data_local_dir;
 use ethers::prelude::Address;
-use std::{env, path::PathBuf, time::Duration};
+use serde::{Deserialize, Serialize};
+use std::{env, num::ParseIntError, path::PathBuf, str::FromStr, time::Duration};
 use structopt::StructOpt;
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, PartialOrd)]
+pub struct Confirmations(u64);
+
+/// A number of Ethereum block confirmations. The minimum value is 1 which
+/// corresponds to accepting a block immediately when first seen.
+impl Confirmations {
+    /// Get the latest confirmed block number with the current number of confirmations.
+    pub fn latest_confirmed_block_number(&self, latest_block_number: u64) -> Option<u64> {
+        if latest_block_number > (self.0 - 1) {
+            Some(latest_block_number - (self.0 - 1))
+        } else {
+            None
+        }
+    }
+}
+
+impl Default for Confirmations {
+    fn default() -> Self {
+        Self(1)
+    }
+}
+
+impl FromStr for Confirmations {
+    type Err = ParseIntError;
+    fn from_str(s: &str) -> Result<Self, ParseIntError> {
+        let n = s.parse::<u64>()?;
+        if n == 0 {
+            panic!("Confirmations must be >= 1")
```

Comment:
StructOpt produces better exit messages if you return the error, rather than invoking the panic!

Reply:
I don't know how to create a `ParseIntError`. Is it impossible? https://stackoverflow.com/a/55616075

Or do you mean defining my own error type for this instead?

Reply:
I meant define an error type to emit instead. But there's actually an exiting type you could use instead.

https://doc.rust-lang.org/std/num/struct.NonZeroU32.html#impl-FromStr

Somehow, this still ends up producing a ParseIntError.